data/src/history.rs

```diff
@@ -49,69 +50,136 @@ impl From<message::Target> for Kind {
     }
 }
 
-pub async fn load(server: &server::Server, kind: &Kind) -> Result<Vec<Message>, Error> {
-    let path = path(server, kind).await?;
+impl From<String> for Kind {
+    fn from(target: String) -> Self {
+        Kind::from(target.as_ref())
+    }
+}
+
+impl From<&str> for Kind {
+    fn from(target: &str) -> Self {
+        if proto::is_channel(target) {
+            Kind::Channel(target.to_string())
+        } else {
+            Kind::Query(target.to_string().into())
+        }
+    }
+}
 
-    Ok(read_all(&path).await.unwrap_or_default())
+pub async fn load_messages(server: &server::Server, kind: &Kind) -> Result<Vec<Message>, Error> {
+    let path = messages_path(server, kind).await?;
+
+    Ok(read_messages(&path).await.unwrap_or_default())
+}
+
+pub async fn load_read_marker(
+    server: server::Server,
+    kind: Kind,
+) -> Result<Option<DateTime<Utc>>, Error> {
```

Comment:
I can imagine a scenario where we have additional metadata related to a buffers history we will want to persist 

Maybe we instead have a struct called Metadata and the first field will be this read marker, but it gives us room to expand it in the future without having to add new files.